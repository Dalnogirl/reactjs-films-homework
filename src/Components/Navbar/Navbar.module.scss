.navbar {
  height: 50px;
  box-shadow: 0 5px 15px 1px rgba(221, 221, 221, 0.5);
  width: 100%;
  display: flex;
  position: relative;
  z-index: 2;
  align-items: center;
  background-color: white;
  color: black;
  //box-shadow: 0 10px 5px 1px black;

  justify-content: space-around;

  .navLink {
    color: black;
    text-decoration: none;
    font-size: 20px;
    transition: all 0.3s ease-in-out;
  }

  .selected {
    color: black;
    font-weight: bold;
    transform: scale(1.2);
  }
}