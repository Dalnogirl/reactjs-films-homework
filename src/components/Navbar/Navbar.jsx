import React, { useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import styles from './Navbar.module.scss'

import {
  moviesActions,
  setGenresObj,
  setMovies
} from '../../redux/moviesReducer'
import Dropdown from '../Dropdown/Dropdown'

const linksArray = ['top_rated', 'popular', 'upcoming']

const Navbar = () => {
  const dispatch = useDispatch()
  const history = useHistory()

  useEffect(() => {
    dispatch(setGenresObj())
  }, [])
  return (
    <div className={styles.navbar}>
      {linksArray.map((item, index) => (
        <div
          key={index}
          data-testid='nav-item'
          className={styles.navLink}
          onClick={() => {
            history.push('')
            dispatch(moviesActions.setCurrentPage(1))
            dispatch(setMovies({ filter: item }))
          }}
        >
          {item[0].toUpperCase() + item.replace('_', ' ').slice(1)}
        </div>
      ))}
      <Dropdown />
    </div>
  )
}

export default React.memo(Navbar)
