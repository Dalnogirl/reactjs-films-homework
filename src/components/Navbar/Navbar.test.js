import React from 'react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import Navbar from './Navbar'
import { fireEvent, render } from '@testing-library/react'
import { Provider } from 'react-redux'
import createMockStore from '../../../__mocks__/mockStore'

const mockHistoryPush = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

describe('Navbar', () => {
  const state = {
    moviesData: {
      isFetching: false,
      moviesList: null,
      genresObj: {},
      currentFilter: 'top_rated',
      currentGenre: null,
      currentPage: 1,
      totalResults: null,
    },
    headerData: {
      isHeaderFetching: false,
      trailerKey: null,
    },
  }
  const store = createMockStore(state)
  it('Renders Navbar component', () => {
    const history = createMemoryHistory()
    const { container, getAllByTestId } = render(
      <Provider store={store}>
        <Router history={history}>
          <Navbar />
        </Router>
      </Provider>,
    )
    expect(container).toMatchSnapshot()
    fireEvent.click(getAllByTestId('nav-item')[0])
    expect(mockHistoryPush).toBeCalledTimes(1)
  })
})
