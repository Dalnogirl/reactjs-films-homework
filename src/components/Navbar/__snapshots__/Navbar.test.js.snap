// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Renders Navbar component 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navLink"
      data-testid="nav-item"
    >
      Top rated
    </div>
    <div
      class="navLink"
      data-testid="nav-item"
    >
      Popular
    </div>
    <div
      class="navLink"
      data-testid="nav-item"
    >
      Upcoming
    </div>
    <div
      class="dd_wrapper"
    >
      <div
        class="dd_header"
        data-testid="ddHeader"
      >
        <div
          class="dd_header_title"
        >
          Search By Genre â–¼
        </div>
      </div>
    </div>
  </div>
</div>
`;
